src/control/lib/netdetect/netdetect.go

```diff
@@ -212,6 +211,50 @@ func (da *DeviceAffinity) String() string {
 	return fmt.Sprintf("%s:%s:%s:%d", da.DeviceName, da.CPUSet, da.NodeSet, da.NUMANode)
 }
 
+type hwlocProtectedAccess struct {
+	mutex sync.Mutex
+}
+
+var hpa hwlocProtectedAccess
+
+func (hpa *hwlocProtectedAccess) hwloc_get_proc_cpubind_wrapper(topology C.hwloc_topology_t, pid C.hwloc_pid_t, cpuset C.hwloc_cpuset_t, flags C.int) C.int {
```

Comment:
I'd be in favor of more Go-like names (GetProcCPUBind here, for example).

Reply:
Ok, I've changed the name.